### ISSUE TITLE
Fix BooleanExpressionComplexity check violations in Checkstyle code

### ISSUE THREAD

**Issue by mkordas** [checkstyle/checkstyle#1052]
Follow-up to #945.

http://checkstyle.sourceforge.net/config_metrics.html#BooleanExpressionComplexity

Let's decide whether we want to fix/prevent code with high boolean complexity. Currently the highest complexity (16) can be found in [FinalLocalVariableCheck](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java#L178):
```java
if ((TokenTypes.POST_DEC == parentType
        || TokenTypes.DEC == parentType
        || TokenTypes.POST_INC == parentType
        || TokenTypes.INC == parentType
        || TokenTypes.ASSIGN == parentType
        || TokenTypes.PLUS_ASSIGN == parentType
        || TokenTypes.MINUS_ASSIGN == parentType
        || TokenTypes.DIV_ASSIGN == parentType
        || TokenTypes.STAR_ASSIGN == parentType
        || TokenTypes.MOD_ASSIGN == parentType
        || TokenTypes.SR_ASSIGN == parentType
        || TokenTypes.BSR_ASSIGN == parentType
        || TokenTypes.SL_ASSIGN == parentType
        || TokenTypes.BXOR_ASSIGN == parentType
        || TokenTypes.BOR_ASSIGN == parentType
        || TokenTypes.BAND_ASSIGN == parentType)
        && ast.getParent().getFirstChild() == ast)
{
    removeVariable(ast);
}
```

All violations:
```
api\JavadocTagInfo.java:73:13: Boolean expression complexity is 4 (max allowed is 3).
api\JavadocTagInfo.java:91:13: Boolean expression complexity is 8 (max allowed is 3).
api\JavadocTagInfo.java:113:13: Boolean expression complexity is 8 (max allowed is 3).
api\JavadocTagInfo.java:134:13: Boolean expression complexity is 9 (max allowed is 3).
api\JavadocTagInfo.java:189:13: Boolean expression complexity is 8 (max allowed is 3).
api\JavadocTagInfo.java:210:13: Boolean expression complexity is 8 (max allowed is 3).
api\JavadocTagInfo.java:231:13: Boolean expression complexity is 8 (max allowed is 3).
api\JavadocTagInfo.java:288:13: Boolean expression complexity is 8 (max allowed is 3).
api\JavadocTagInfo.java:330:13: Boolean expression complexity is 6 (max allowed is 3).
api\JavadocTagInfo.java:370:13: Boolean expression complexity is 8 (max allowed is 3).
api\JavadocTagInfo.java:407:13: Boolean expression complexity is 8 (max allowed is 3).
api\JavadocTagInfo.java:429:13: Boolean expression complexity is 4 (max allowed is 3).
checks\AvoidEscapedUnicodeCharactersCheck.java:216:9: Boolean expression complexity is 7 (max allowed is 3).
checks\DeclarationCollector.java:410:13: Boolean expression complexity is 4 (max allowed is 3).
checks\UncommentedMainCheck.java:178:9: Boolean expression complexity is 4 (max allowed is 3).
checks\annotation\AnnotationLocationCheck.java:252:9: Boolean expression complexity is 4 (max allowed is 3).
checks\annotation\AnnotationUseStyleCheck.java:455:14: Boolean expression complexity is 4 (max allowed is 3).
checks\coding\EqualsHashCodeCheck.java:94:9: Boolean expression complexity is 4 (max allowed is 3).
checks\coding\ExplicitInitializationCheck.java:152:9: Boolean expression complexity is 5 (max allowed is 3).
checks\coding\FinalLocalVariableCheck.java:178:17: Boolean expression complexity is 16 (max allowed is 3).
checks\coding\HiddenFieldCheck.java:298:13: Boolean expression complexity is 7 (max allowed is 3).
checks\coding\IllegalInstantiationCheck.java:290:13: Boolean expression complexity is 4 (max allowed is 3).
checks\coding\RequireThisCheck.java:185:9: Boolean expression complexity is 7 (max allowed is 3).
checks\design\VisibilityModifierCheck.java:547:47: Boolean expression complexity is 10 (max allowed is 3).
checks\design\VisibilityModifierCheck.java:636:22: Boolean expression complexity is 4 (max allowed is 3).
checks\imports\CustomImportOrderCheck.java:510:9: Boolean expression complexity is 4 (max allowed is 3).
checks\imports\CustomImportOrderCheck.java:595:9: Boolean expression complexity is 4 (max allowed is 3).
checks\indentation\SlistHandler.java:111:9: Boolean expression complexity is 11 (max allowed is 3).
checks\javadoc\JavadocMethodCheck.java:456:9: Boolean expression complexity is 4 (max allowed is 3).
checks\javadoc\JavadocMethodCheck.java:493:9: Boolean expression complexity is 4 (max allowed is 3).
checks\javadoc\JavadocParagraphCheck.java:164:13: Boolean expression complexity is 4 (max allowed is 3).
checks\javadoc\JavadocStyleCheck.java:192:9: Boolean expression complexity is 6 (max allowed is 3).
checks\javadoc\JavadocTag.java:136:9: Boolean expression complexity is 4 (max allowed is 3).
checks\javadoc\JavadocTypeCheck.java:235:9: Boolean expression complexity is 6 (max allowed is 3).
checks\javadoc\JavadocVariableCheck.java:172:9: Boolean expression complexity is 4 (max allowed is 3).
checks\javadoc\TagParser.java:150:9: Boolean expression complexity is 4 (max allowed is 3).
checks\metrics\JavaNCSSCheck.java:201:9: Boolean expression complexity is 4 (max allowed is 3).
checks\modifier\RedundantModifierCheck.java:103:17: Boolean expression complexity is 4 (max allowed is 3).
checks\naming\AbstractAccessControlNameCheck.java:102:9: Boolean expression complexity is 7 (max allowed is 3).
checks\naming\ConstantNameCheck.java:89:9: Boolean expression complexity is 5 (max allowed is 3).
checks\whitespace\EmptyLineSeparatorCheck.java:252:21: Boolean expression complexity is 4 (max allowed is 3).
checks\whitespace\GenericWhitespaceCheck.java:171:22: Boolean expression complexity is 7 (max allowed is 3).
checks\whitespace\WhitespaceAroundCheck.java:415:9: Boolean expression complexity is 7 (max allowed is 3).
checks\whitespace\WhitespaceAroundCheck.java:497:13: Boolean expression complexity is 4 (max allowed is 3).
filters\SuppressElement.java:146:9: Boolean expression complexity is 6 (max allowed is 3).
```

**Comment by romani** [comment]
We need to change that Check to consider sequential comparison by the same operator and count it as "1" point to complexity.

example in description should produce complexity - 2 .

**Comment by romani** [comment]
This Check is non-practical in his current form, I would rather update his default value to 7.
All cases above 7 need to be resolved, as it clear evidence that too much is that condition.

case from issue description, and from JavadocTagInfo.java, could be refactoring to array container and search over array.

**Comment by OldIMP** [comment]
> We need to change that Check to consider sequential comparison by the same operator and the same functions and count it as "1" point to complexity (I know that is uncertain rule for now).
> 
> example in description should produce complexity - 2 .
> 
> I would make a special issue on that, as it require a lot of investigations.

Would you guys still consider implementing this or just stay w/ a higher default(7)? The former would be really preferable since it's common to have a long `&&` chain in unit tests.

**Comment by rnveach** [comment]
@OldIMP As the quoted text says, if the issue is still exists, then you can always create a new issue on it. I am not aware of any outstanding issues for this functionality but you can try searching first.

**Comment by OldIMP** [comment]
> We need to change that Check to consider sequential comparison by the same operator and the same functions and count it as "1" point to complexity (I know that is uncertain rule for now).
> 
> example in description should produce complexity - 2 .
> 
> I would make a special issue on that, as it require a lot of investigations.

Found #1525, which has been open for a while.
